This file was auto-generated on 2020-02-10 12:06:38.068079

# Subject Fields
## Age
### The age of the subject in years.
```
age
```
## Gender
### The gender of the subject.
```
sex
gender
```
## IPOP Response - Three Type
### The response of the subject encoded as a three-type variable.
```
three-type response
ipop three-type response
```
## IPOP Response - Two Type
### The response of the subject encoded as a two-type variable.
```
two-type response
ipop two-type response
```
## RECIST
### The RECIST classification of the subject.
```
recist
```
## Raw Investigator Response
### The response of the subject encoded by the investigator.
```
raw investigator response
raw response
```
## Stage
### The tumor stage of the subject.
```
stage
m stage
metastasis stage
```
# Sample Fields
## ADC
```
adc
```
## AMP2
```
apm2
```
## Aging Signature
### The aging signature (Sanger signature 1) of the sample.
```
aging signature
```
## Angiogenesis
```
angiogenesis
```
## B-Cells
```
b cells
```
## B-Cells Memory
```
b cells memory
```
## B-Cells Naive
```
t cells follicular helper
```
## CD8 T-Cells
```
cd8 t cells
```
## Cancer Type
### The type of cancer of the sample.
```
cancer type
type of cancer
```
## Cancer Type Detailed
### The detailed type of cancer of the sample.
```
cancer type detailed
detailed cancer type
detailed type of cancer
```
## Cytotoxic Cells
```
cytotoxic cells
```
## DC
```
dc
```
## Davoli SCNA
```
davoli scna
```
## Dentritic Cells Activated
```
dendritic cells activated
```
## Fraction Genome Altered
```
fraction genome altered
```
## Fraction Genome CNA
```
fraction genome cna
```
## Gene Panel
### The gene panel taken using the sample.
```
gene panel
```
## HLA A Allele
```
hla a allele
```
## HLA A Evolutionary Diversity
```
hla a evolutionary diversity
hed a
```
## HLA A Zygosity
```
hla a zygosity
hla a
```
## HLA B Allele
```
hla b allele
```
## HLA B Evolutionary Diversity
```
hla b evolutionary diversity
hed b
```
## HLA B Zygosity
```
hla b zygosity
hla b
```
## HLA C Allele
```
hla c allele
```
## HLA C Evolutionary Diversity
```
hla c evolutionary diversity
hed c
```
## HLA C Zygosity
```
hla c zygosity
hla c
```
## HLA Mean Evolutionary Diversity
```
hla mean evolutionary diversity
hed mean
mean hed
```
## HLA Zygosity
```
hla zygosity
hla
```
## IDC
```
idc
```
## Indel Count
### The number of insertions and deletions (indels) detected and validated by an internal pipeline of the sample.
```
indel count
```
## Institute
### The institute associated with the sample.
```
institute
```
## Log Neoantigen Count
### The logarithm (base 10) of the number of non-synonmous single-nucleotide variants (SNV) predicted to encode neo-antigenic peptides of a sample. See Neoantigen Count for more information. In the instance that the number of neoantigens is 0, this field is set to the logarithm of 1 (base 10), which equals 0.
```
log neoantigen count
```
## Log Neopeptide Count
### The logarithm (base 10) of the number of non-synonmous single-nucleotide variant-derived neopeptides predicted to encode neo-antigenic peptides of a sample. See Neopeptide Count for more information. In the instance that the number of neopeptides is 0, this field is set to the logarithm of 1 (base 10), which equals 0.
```
log neopeptide count
```
## Log Tumor Mutation Burden
### The logarithm (base10) of the normalized number of non-synonmous single-nucleotide variants (SNV) detected and validated by an internal pipeline of the sample. In the instance that the non-synonmous SNV count is 0, this field is set to the logarithm of 1/30 (base 10).
```
log tmb
log tumor mutation burden
log mutation burden
```
## Log Tumor Mutation Burden Indel
### The logarithm (base 10) of the normalized number of insertions and deletions (indels) detected and validated by an internal pipeline of the sample. In the instance that the number of indels is 0, this field is set to the logarithm of 1/30 (base 10).
```
log tmb indel
log tumor mutation burden indel
log mutation burden indel
```
## Log Tumor Mutation Burden Syn
### The logarithm (base 10) of the normalized number of single-nucleotide variants (SNV) detected and validated by an internal pipeline of the sample. In the instance that the SNV count is 0, this field is set to the logarithm of 1/30 (base 10).
```
log tmb syn
log tumor mutation burden syn
log mutation burden syn
```
## MSI Score
### The microsatellite instability score of the sample.
```
msi score
```
## MSI Type
### The microsatellite instability type of the sample.
```
msi type
```
## Macrophages
```
macrophages
```
## Macrophages M0
```
macrophages m0
```
## Macrophages M1
```
macrophages m1
```
## Macrophages M2
```
macrophages m2
```
## Monocytes
```
monocytes
```
## Mutation Count
### The number of non-synonmous single-nucleotide variants (SNV) detected and validated by an internal pipeline of the sample.
```
mutation count
number of mutations
```
## Mutation Count with Synonymous Mutations
### The number of non-synonmous single-nucleotide variants (SNV) detected and validated by an internal pipeline of the sample.
```
mutation count syn
```
## NK CD56 Bright Cells
```
nk cd56 bright cells
```
## NK Cells
```
nk cells
```
## Natural Killer Activated
```
natural killer activated
```
## Natural Killer Resting
```
natural killer resting
```
## Neoantigen Count
### The number of non-synonmous single-nucleotide variants (SNV) predicted to encode neo-antigenic peptides of a sample. The prediction is made by NetMHCPan and the length of amino acids is 9.
```
neoantigen count
```
## Neopeptide Count
### The number of non-synonmous single-nucleotide variant-derived neopeptides predicted to encode neo-antigenic peptides of a sample. The prediction is made by NetMHCPan and the length of amino acids is 9.
```
neopeptide count
```
## Neutrophils Cibersort
```
neutrophils cibersort
```
## Neutrophils SSGSEA
```
neutrophils ssgsea
```
## Oncotree Code
### The OncoTree code of the sample.
```
oncotree
oncotree code
```
## PD1
```
pd1
```
## PDL1
```
pdl1
```
## Plasma
```
plasma
```
## Sample Class
### The class of sample.
```
sample class
class of sample
```
## Sample Cover
### The cover of the sample.
```
sample cover
```
## Sample Type
### The type of sample.
```
sample type
type of sample
```
## Smoking Signature
### The smoking signature (Sanger signature 4) of the sample.
```
smoking signature
```
## Somatic Status
### The somatic status of the sample.
```
somatic status
```
## T Helper Cells
```
t helper cells
```
## T-Cells
```
t cells
```
## T-Cells CD4 Naive
```
t cells cd4 naive
```
## T-Cells CD4 Resting
```
t cells cd4 resting
```
## T-Cells CD8
```
t cells cd8
```
## T-Cells Regulator
```
t cells regulator
```
## TCM Cells
```
tcm cells
```
## TEM Cells
```
tem cells
```
## TFH Cells
```
tfm cells
```
## TGD Cells
```
tgd cells
```
## TH1 Cells
```
th1
```
## TH17 Cells
```
th17 cells
```
## TH2 Cells
```
th2 cells
```
## TREG Cells
```
treg cells
```
## Treatment Status
### The treatment status at the time the sample was taken.
```
treatment status
pre on
```
## Tumor Mutation Burden
### The normalized number of non-synonmous single-nucleotide variants (SNV) detected and validated by an internal pipeline of the sample. To normalize the mutation count, the number of non-synonmous SNVs is divided by 30.
```
tmb
tumor mutation burden
mutation burden
```
## Tumor Mutation Burden of Indel Mutations
### The normalized number of insertions and deletions (indels) detected and validated by an internal pipeline of the sample. To normalize the mutation count, the number of indels is divided by 30,
```
tmb syn
tumor mutation burden syn
mutation burden syn
```
## Tumor Mutation Burden with Synonymous Mutations
### The normalized number of single-nucleotide variants (SNV) detected and validated by an internal pipeline of the sample. To normalize the mutation count, the number of SNVs is divided by 30.
```
tmb syn
tumor mutation burden syn
mutation burden syn
```
## Tumor Purity
### The tumor purity of the sample.
```
tumor purity
```
## UV Signature
### The UV signature (Sanger signature 7) of the sample.
```
uv signature
```
